urlfd018a75f7_highlanderkev_StockExVirtuoso_tgz-pJ8-driver_Phase4J8/src/tradable/Trader.java

```diff
@@ -0,0 +1,337 @@
+     */
+    @Override
+    public int getCancelledVolume() {
+        return cancelledVolume;
+    }
+
+    /**
+     * Private Set method which sets user name of tradable.
+     *
+     * @param newUser
+     * @throws Exception
+     */
+    private void setUser(String newUser) throws Exception {
+        if (ExceptionHandler.checkString(newUser, "tradable.trader#setUser.")) {
+            user = newUser;
+        }
+    }
+
+    /**
+     * Public Get method which returns the current user name.
+     *
+     * @return string of user name
+     */
+    @Override
+    public String getUser() {
+        return user;
+    }
+
+    /**
+     * Private Set method which sets the side to either BUY or SELL.
+     *
+     * @param book side (BUY/SELL)
+     * @throws Exception
+     */
+    private void setSide(BookSide book) throws Exception {
+        if (ExceptionHandler.checkObject(book, "tradable.Trader#setSide.")) {
+            side = book;
+        }
+    }
+
+    /**
+     * Public Get method which returns the current book side (BUY/SELL).
+     *
+     * @return BookSide
+     */
+    @Override
+    public BookSide getSide() {
+        return side;
+    }
+
+    /**
+     * Private Set method to set (isQuote) boolean.
+     *
+     * @param setQT it is a Quote object
+     */
+    private void setQuote(boolean isQT) {
+        isQuote = isQT;
+    }
+
+    /**
+     * Public Get method returns whether the current object is a quote or not.
+     *
+     * @return boolean if it is a quote object
+     */
+    @Override
+    public boolean isQuote() {
+        return isQuote;
+    }
+
+    /**
+     * Private Set method for System ID.
+     *
+     * @param orderPrice
+     * @param userName
+     * @param productSymbol
+     * @throws Exception
+     */
+    private void setId(Price orderPrice, String userName, String productSymbol) throws Exception {
+        long currentTime = System.nanoTime();
+        String thisClass = "tradable.Trader#setID.";
+        if (ExceptionHandler.checkString(userName, thisClass) && ExceptionHandler.checkString(productSymbol, thisClass)
+                && ExceptionHandler.checkObject(orderPrice, thisClass)) {
+            String orderValue = orderPrice.toString();
+            String newId = (userName + productSymbol + orderValue + currentTime);
+            id = newId;
+        }
+    }
+
+    /**
+     * Public Get method returns current system id of this tradable.
+     *
+     * @return String of System ID
+     */
+    @Override
+    public String getId() {
+        String thisId = id;
+        return thisId;
+    }
+
+    /**
+     * Public To String method to return a string representation of the Tradable
+     * values.
+     *
+     * @return String of order values
+     */
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(String.format("%s order: %s %s %s at %s", user, side, remainingVolume, product, price.toString()));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: remainingVolume. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 332. Consider formatting this data with &#39;%d&#39; instead.